<?xml version="1.0"?>
<launch>
  <arg name="log_level" default="info"/>
  <arg name="rviz" default="false" description="launch rviz"/>
  <arg name="rviz_respawn" default="true" description="respawn rviz"/>
  <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
  <arg name="model_path" default="$(var data_path)/vad"/>
  <arg name="plugins_path" default="$(find-pkg-share autoware_tensorrt_plugins)"/>

  <arg name="camera_info0" default="/sensing/camera/camera0/camera_info"/>
  <arg name="camera_info1" default="/sensing/camera/camera1/camera_info"/>
  <arg name="camera_info2" default="/sensing/camera/camera2/camera_info"/>
  <arg name="camera_info3" default="/sensing/camera/camera3/camera_info"/>
  <arg name="camera_info4" default="/sensing/camera/camera4/camera_info"/>
  <arg name="camera_info5" default="/sensing/camera/camera5/camera_info"/>

  <arg name="image0" default="/sensing/camera/camera0/image_rect_color"/>
  <arg name="image1" default="/sensing/camera/camera1/image_rect_color"/>
  <arg name="image2" default="/sensing/camera/camera2/image_rect_color"/>
  <arg name="image3" default="/sensing/camera/camera3/image_rect_color"/>
  <arg name="image4" default="/sensing/camera/camera4/image_rect_color"/>
  <arg name="image5" default="/sensing/camera/camera5/image_rect_color"/>

  <arg name="kinematic_state" default="/localization/kinematic_state"/>
  <arg name="acceleration" default="/localization/acceleration"/>

  <arg name="output/objects" default="perception/object_recognition/vad/objects"/>
  <arg name="output/trajectory" default="planning/vad/trajectory"/>
  <arg name="output/trajectories" default="planning/vad/trajectories"/>
  <arg name="output/map" default="debug/vad/map"/>

  <node pkg="autoware_tensorrt_vad" exec="vad_node" name="vad" output="screen" args="--ros-args --log-level $(var log_level)">
    <param from="$(find-pkg-share autoware_tensorrt_vad)/config/vad_tiny.param.yaml" allow_substs="true"/>
    <param from="$(find-pkg-share autoware_tensorrt_vad)/config/ml_package_vad_tiny.param.yaml" allow_substs="true"/>
    <param from="$(find-pkg-share autoware_tensorrt_vad)/config/object_class_remapper.param.yaml" allow_substs="true"/>

    <remap from="~/output/objects" to="$(var output/objects)"/>
    <remap from="~/output/trajectory" to="$(var output/trajectory)"/>
    <remap from="~/output/trajectories" to="$(var output/trajectories)"/>
    <remap from="~/output/map" to="$(var output/map)"/>

    <remap from="~/input/camera_info0" to="$(var camera_info0)"/>
    <remap from="~/input/camera_info1" to="$(var camera_info1)"/>
    <remap from="~/input/camera_info2" to="$(var camera_info2)"/>
    <remap from="~/input/camera_info3" to="$(var camera_info3)"/>
    <remap from="~/input/camera_info4" to="$(var camera_info4)"/>
    <remap from="~/input/camera_info5" to="$(var camera_info5)"/>
    <remap from="~/input/image0" to="$(var image0)"/>
    <remap from="~/input/image1" to="$(var image1)"/>
    <remap from="~/input/image2" to="$(var image2)"/>
    <remap from="~/input/image3" to="$(var image3)"/>
    <remap from="~/input/image4" to="$(var image4)"/>
    <remap from="~/input/image5" to="$(var image5)"/>
    
    <remap from="~/input/kinematic_state" to="$(var kinematic_state)"/>
    <remap from="~/input/acceleration" to="$(var acceleration)"/>
  </node>
</launch>
